tests/qnn/test_keras.py

```diff
@@ -138,6 +177,52 @@ class FuncPatch:
             with pytest.raises(TypeError, match="Cannot have a variable number of keyword"):
                 KerasLayer(c, w, output_dim)
 
+    @pytest.mark.parametrize("n_qubits, output_dim", indices_up_to(1))
+    def test_var_keyword_tape_mode(self):
+        """Test that variable number of keyword arguments works in tape mode"""
+        if not qml.tape_mode_active():
+            pytest.skip("This functionality is only supported in tape mode.")
+
+        n_qubits = 2
+        output_dim = 2
+
+        dev = qml.device("default.qubit", wires=n_qubits)
+        w = {
+            "w1": (3, n_qubits, 3),
+            "w2": (1,),
+            "w3": 1,
+            "w4": [3],
+            "w5": (2, n_qubits, 3),
+            "w6": 3,
+            "w7": 0,
+        }
+
+        @qml.qnode(dev, interface="tf")
+        def c(inputs, **kwargs):
+            """A circuit that embeds data using the AngleEmbedding and then performs a variety of
+            operations. The output is a PauliZ measurement on the first output_dim qubits. One set of
+            parameters, w5, are specified as non-trainable."""
+            qml.templates.AngleEmbedding(inputs, wires=list(range(n_qubits)))
+            qml.templates.StronglyEntanglingLayers(kwargs["w1"], wires=list(range(n_qubits)))
+            qml.RX(kwargs["w2"][0], wires=0 % n_qubits)
+            qml.RX(kwargs["w3"], wires=1 % n_qubits)
+            qml.Rot(*kwargs["w4"], wires=2 % n_qubits)
+            qml.templates.StronglyEntanglingLayers(kwargs["w5"], wires=list(range(n_qubits)))
+            qml.Rot(*kwargs["w6"], wires=3 % n_qubits)
+            qml.RX(kwargs["w7"], wires=4 % n_qubits)
+            return [qml.expval(qml.PauliZ(i)) for i in range(output_dim)]
+
+        layer = KerasLayer(c, w, output_dim=output_dim)
+        x = tf.ones((2, n_qubits))
+
+        layer_out = layer(x)
+        circ_weights = layer.qnode_weights.copy()
+        circ_weights["w4"] = 1.0 * circ_weights["w4"]
+        circ_weights["w6"] = 1.0 * circ_weights["w6"]
```

Comment:
This is so that we don't have problems with `Variable` slicing in TF (also used a few more times below)